R/proDA.R

```diff
@@ -313,36 +313,46 @@ fit_parameters_loop <- function(Y, model_matrix, location_prior_df,
   }
   # Initialization
   n_samples <- ncol(Y)
-
-
+  mM_t <- t(model_matrix)
+  seq_Y <- seq_len(nrow(Y))
+  seq_mM <- seq_len(nrow(model_matrix))
+  
   Y_compl <- Y
-  Y_compl[is.na(Y)] <- rnorm(sum(is.na(Y)), mean=quantile(Y, probs=0.1, na.rm=TRUE), sd=sd(Y,na.rm=TRUE)/5)
-  res_init <- lapply(seq_len(nrow(Y)), function(i){
-    pd_lm.fit(Y_compl[i, ], model_matrix,
+  y_na <- is.na(Y)
+  Y_compl[y_na] <- rnorm(sum(y_na), mean=quantile(Y, probs=0.1, na.rm=TRUE), sd=sd(Y,na.rm=TRUE)/5)
+  
+  split_Y_compl <- split(Y_compl, factor(rownames(Y_compl), rownames(Y_compl)))
+  names(split_Y_compl) <- NULL
+  res_init <- lapply(split_Y_compl, function(i){
```

Comment:
Can you choose a more descriptive variable name. For me `i` always is an index. I think for example `y_i` would be better :)